tests/test_rhymes.py

```diff
@@ -470,3 +490,605 @@ def test_analyze_rhyme_romance(romance):
+    sobre la frente triste.
+    A lo lejos alzábanse los muros
+    de la ciudad teológica, en que vive
+    la sempiterna Paz. La negra barca
+    llegó a la ansiada costa, y el sublime
+    espíritu gozó la suma gracia;
+    y ¡oh Montaigne! Núñez vio la cruz erguirse,
+    y halló al pie de la sacra Vencedora
+    el helado cadáver de la Esfinge."""
+    output = "silva_arromanzada"
+    input_poem = get_scansion(poem, rhyme_analysis=True)
+    assert input_poem[0]["structure"] == output
+
+
+def test_rhyme_cantar():
+    poem = """Algunos desesperados
+    solo se curan con soga;
+    otros, con siete palabras:
+    la fe se ha puesto de moda."""
+    output = "cantar"
+    input_poem = get_scansion(poem, rhyme_analysis=True)
+    assert input_poem[0]["structure"] == output
+
+
+def test_rhyme_lira():
+    poem = """Si de mi baja lira
+    tanto pudiese el son que en un momento
+    aplacase la ira
+    del animoso viento
+    y la furia del mar y el movimiento."""
+    output = "lira"
+    input_poem = get_scansion(poem, rhyme_analysis=True)
+    assert input_poem[0]["structure"] == output
+
+
+def test_get_stressed_endings_sinalepha():
+    # aplacase la ira
+    lines = [
+        {'tokens': [
+            {'word': [
+                {'syllable': 'a', 'is_stressed': False},
+                {'syllable': 'pla', 'is_stressed': False},
+                {'syllable': 'ca', 'is_stressed': True},
+                {'syllable': 'se', 'is_stressed': False,
+                 'is_word_end': True}],
+                'stress_position': -2},
+            {'word': [
+                {'syllable': 'la',
+                 'is_stressed': False,
+                 'has_synalepha': True,
+                 'is_word_end': True}],
+                'stress_position': 0},
+            {'word': [
+                {'syllable': 'i', 'is_stressed': True},
+                {'syllable': 'ra', 'is_stressed': False,
+                 'is_word_end': True}],
+                'stress_position': -2}],
+            'phonological_groups': [
+                {'syllable': 'a', 'is_stressed': False},
+                {'syllable': 'pla', 'is_stressed': False},
+                {'syllable': 'ca', 'is_stressed': True},
+                {'syllable': 'se', 'is_stressed': False,
+                 'is_word_end': True},
+                {'syllable': 'lai', 'is_stressed': True,
+                 'synalepha_index': [1]},
+                {'syllable': 'ra', 'is_stressed': False,
+                 'is_word_end': True}],
+            'rhythm': {'stress': '--+-+-', 'type': 'pattern', 'length': 6}}
+    ]
+    output = [
+        (["i", "ra"], 6, -2)
+    ]
+    assert get_stressed_endings(lines) == output
+
+
+def test_seguidilla():
+    poem = """Que se caiga la torre
+    de Valladolid
+    como a mí no me coja,
+    ¿qué se me da a mí?"""
+    output = "seguidilla"
+    input_poem = get_scansion(poem, rhyme_analysis=True)
+    assert input_poem[0]["structure"] == output
```

Comment:
Maybe put all the specific structure tests in its own test file.

Reply:
I'll add an issue for that and work on this later on